### PR TITLE
Add documentation for removing auto-provisioned toolchains

### LINKED ISSUE #24581: Provide documentation about removing auto-provisioned toolchains
Gradle can [auto-provision toolchains](https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning).

There is documentation on that, but no instructions about how to properly remove an auto-provisioned toolchain.

A note about stopping gradle daemon should be added at least (to clear the cache)

### PR DESCRIPTION
**Pull request by anazworth**
Added instructions to properly remove an auto-provisioned toolchain and a note about stopping the Gradle Daemon to clear the cache

Documentation change only.

Fixes #24581 

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/jvm/toolchains.adoc [hunk 1]
```diff
@@ -330,6 +330,16 @@ In order to disable auto-provisioning, you can use the `org.gradle.java.installa
 * Either start gradle using `-Porg.gradle.java.installations.auto-download=false`
 * Or put `org.gradle.java.installations.auto-download=false` into a `gradle.properties` file.
 
+[[sub:remove_auto_provisioned_toolchain]]
+=== How to remove an auto-provisioned toolchain
+
+To properly remove an auto-provisioned toolchain, you can remove or modify any reference to the toolchain from `settings.gradle.kts` or `settings.gradle`
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/jvm/toolchains.adoc:336]:
```suggestion
To properly remove an auto-provisioned toolchain, you can remove or modify any reference to the toolchain from `settings.gradle.kts` or `settings.gradle`.
```

**Review by lkasso — APPROVED** [review]

**Comment by ov7a** [comment]
@anazworth Thank you for your interest in Gradle!

Please note that the original issue was about files on the disc, not the build script.

**Review by ov7a — CHANGES_REQUESTED** [review]
@anazworth Thank you for your interest in Gradle!

Please note that the original issue was about files on the disc, not the build script.

**Review by ov7a — CHANGES_REQUESTED** [review]

File: subprojects/docs/src/docs/userguide/jvm/toolchains.adoc [hunk 2]
```diff
@@ -330,6 +330,18 @@ In order to disable auto-provisioning, you can use the `org.gradle.java.installa
 * Either start gradle using `-Porg.gradle.java.installations.auto-download=false`
 * Or put `org.gradle.java.installations.auto-download=false` into a `gradle.properties` file.
 
+[[sub:removing_auto_provisioned_toolchain]]
+=== Removing an auto-provisioned toolchain
+
+There are cases where a user might want to take explicit control over the toolchain configuration for their project. This could include specifying custom versions of a toolchain, ensuring compatibility with specific project requirements, or addressing issues related to the auto-provisioned toolchain.
+
+In a situation where it's necessary to remove an auto-provisioned toolchain, remove the relevant toolchain located in the `/jdks` directory within the <<directory_layout.adoc#dir:gradle_user_home,Gradle User Home>>.
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/jvm/toolchains.adoc:338]:
```suggestion
When removing an auto-provisioned toolchain is necessary, remove the relevant toolchain located in the `/jdks` directory within the <<directory_layout.adoc#dir:gradle_user_home,Gradle User Home>>.
```

**Review by lkasso — DISMISSED** [review]
Minor english cleanup.

File: subprojects/docs/src/docs/userguide/jvm/toolchains.adoc [hunk 3]
```diff
@@ -330,6 +330,19 @@ In order to disable auto-provisioning, you can use the `org.gradle.java.installa
 * Either start gradle using `-Porg.gradle.java.installations.auto-download=false`
 * Or put `org.gradle.java.installations.auto-download=false` into a `gradle.properties` file.
 
+[[sub:removing_auto_provisioned_toolchain]]
+=== Removing an auto-provisioned toolchain
+
+In some cases, a user might want explicit control over the toolchain configuration for their project.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/jvm/toolchains.adoc:336]: This line and the next one do not provide any new information.
The section is in the toolchains chapter so there is no need to re-iterate what toolchains provide.

Please remove this line and the next one.

**Review by ljacomet — CHANGES_REQUESTED** [review]
See the comment below.

The documentation describing what to delete and the associated note are good to go after that.

**Comment by lkasso** [comment]
Reopening to clean up docs

**Comment by lkasso** [comment]
@anazworth, I merged your changes in https://github.com/gradle/gradle/pull/28972.
Thank you for your contribution!

**Pull request closed without merging**